### PR TITLE
Cleanup DAML Script over JSON API

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala [hunk 1]
```diff
@@ -1078,10 +1060,9 @@ object JsonLedgerClient {
...
         "key" -> LfValueCodec.apiValueToJsValue(args.key))
     implicit val fetchReader: RootJsonReader[FetchResponse] = v =>
-      v.asJsObject.getFields("result") match {
-        case Seq(JsNull) => FetchResponse(None)
-        case Seq(r) => FetchResponse(Some(r.convertTo[ActiveContract]))
-        case _ => deserializationError(s"Could not parse FetchResponse: $v")
+      v match {
+        case JsNull => FetchResponse(None)
+        case r => FetchResponse(Some(r.convertTo[ActiveContract]))
     }
```
Comment (S11001001) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala:1066]:
```suggestion
      FetchResponse(v.convertTo[Option[ActiveContract]])
```

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala [hunk 2]
```diff
@@ -1133,39 +1104,30 @@ object JsonLedgerClient {
...
-                        case _ =>
-                          deserializationError(s"Could not parse CreateAndExerciseResponse: $v")
-                      }
-                    case _ => deserializationError(s"Could not parse CreateAndExerciseResponse: $v")
+    implicit val createAndExerciseReader: RootJsonReader[CreateAndExerciseResponse] = v =>
+      v.asJsObject.getFields("exerciseResult", "events") match {
+        case Seq(result, events) =>
+          events match {
+            case JsArray(events) if events.size >= 1 =>
+              events.head.asJsObject.getFields("created") match {
```
Comment (S11001001) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala:1112]: Maybe we should ban `head` 🔨 (I know just a reindent but...)

```suggestion
            case JsArray(Seq(event, _*)) =>
              event.asJsObject.getFields("created") match {
```

Comment (cocreature) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala:1109]: good catch, I didn’t pay attention to that.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala (hunk 1 above)
Comment (cocreature) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala]: annoyingly, spray only provides a JsonFormat for Option which means we also need a JsonFormat (and thereby a writer for) ActiveContrct. I’ve just written my own reader-only implicit for Option now.

**Review by S11001001 — APPROVED** [review]

**Pull request merged**